### PR TITLE
Fix GotoDefinitionHelpers.TryGoToDefinition helper method to not assu…

### PR DESCRIPTION
**Pull request by mavasani**
…me that the document for the resultant definition is within the same workspace as the original document. An external result provider might return a navigable item with location in a document in another workspace.

### CODE REVIEW

**Comment by mavasani** [comment]
@srivatsn @heejaechang @dotnet/roslyn-analysis @dotnet/roslyn-ide can you please review?

This is to unblock GotoDefinition from Roslyn VS workspace to an external workspace.

File: src/EditorFeatures/Core/Implementation/GoToDefinition/GoToDefinitionHelpers.cs [hunk 1]
```diff
@@ -98,7 +99,18 @@ internal static class GoToDefinitionHelpers
         public static bool TryGoToDefinition(
             ImmutableArray<INavigableItem> navigableItems,
             string title,
-            Workspace workspace,
+            Workspace originalWorkspace,
```
Comment (heejaechang) [inline src/EditorFeatures/Core/Implementation/GoToDefinition/GoToDefinitionHelpers.cs]: there is only 1 workspace, so why it is original workspace, what is other workspace?

Comment (heejaechang) [inline src/EditorFeatures/Core/Implementation/GoToDefinition/GoToDefinitionHelpers.cs]: if this is just to change option temporary, how about something like TryGoToDefintionWithProvisionalTabIfPossible and give option in?

Comment (mavasani) [inline src/EditorFeatures/Core/Implementation/GoToDefinition/GoToDefinitionHelpers.cs]: I meant origin workspace where the GTD was invoked. Let me fix the name

**Comment by heejaechang** [comment]
is this due to the option is internal?

**Comment by heejaechang** [comment]
:+1:

File: src/EditorFeatures/Core/Implementation/GoToDefinition/GoToDefinitionHelpers.cs (hunk 1 above)
Comment (heejaechang) [inline src/EditorFeatures/Core/Implementation/GoToDefinition/GoToDefinitionHelpers.cs]: I see, but still this doesn't require anything from workspace except option, so why not just pass in option?

Comment (mavasani) [inline src/EditorFeatures/Core/Implementation/GoToDefinition/GoToDefinitionHelpers.cs]: That seems better, let me make that change.

**Comment by jasonmalinowski** [comment]
:+1:

**Pull request merged**